txscript/scriptnum.go

```diff
@@ -223,3 +223,56 @@ func makeScriptNum(v []byte, requireMinimal bool, scriptNumLen int) (scriptNum,
 
 	return scriptNum(result), nil
 }
+
+// minimallyEncode takes in a byte slice and returns a slice that contains a
+// minimally encoded version of the original.
+func minimallyEncode(data []byte) (ret []byte) {
+	if len(data) == 0 {
+		return []byte{}
+	}
+
+	// If the last byte is not 0x00 or 0x80, we are minimally encoded.
+	last := data[len(data)-1]
+	if last&0x7f > 0 {
+		ret = make([]byte, len(data))
+		copy(ret, data)
+		return
+	}
+
+	// If the script is one byte long, then we have a zero, which encodes as an
```

Comment:
"If the script is one byte long," comment doesn't make sense but that's what ABC has lol